fdk/http_stream.py

```diff
@@ -13,85 +13,67 @@
 #    under the License.
 
 import asyncio
+import h11
 import os
 
-from aiohttp import web
-
 from fdk import constants
 from fdk import log
 from fdk import runner
 
 
 def handle(handle_code):
-    async def pure_handler(request):
+    async def pure_handler(request_reader, response_writer):
+        s = h11.Connection(h11.SERVER)
+
+        raw_data = await request_reader.read(10485760)
```

Comment:
Unfortunately it’s nevessary. H11 accepts list of bytes, but not a raw stream, so, someone needs to read the data (this is what you get while you use asyncio protocols, by the downside of a protocols that you can’t start async operation within that, because you don’t have an access to an event loop.

Reply:
i think we can probably only receive 64K in socket recv without using some fancy features? (I recall horrors from a time when I did socket programming...) `2**16` is it I think, 10MB is huge. this also seems to be the default buffer size limit for the asyncio server request_reader.

we need to handle connection closed and actually close the connection, or we can leak connections. at the moment we're using keep alives and fn is closing idle conns normally, but it could happen one day. do we handle keep alives here? do we handle connection errors (even connection closed) and bail or do we just trot along and call the function with broken data and send a response and move to the next one?

don't really need events_map do we? allocating maps is expensive usually, we just have 2 fields to assign to, and later we just assign the data field to body to assign to data again, let's just do the thing?

we should wait for the conn event to receive data, and only do it if it's h11.NEED_DATA? unclear here, but we could get a different event type, and the way this looks it just reads and goes instead of checking the type. 

may be a good idea to extract this code to a generic http request handler with the h11 stuff, and then we can make a handler that does the fdk thing out of that, this would be a nice separation of concerns and maybe read a bit better and be easier to reason about

Reply:
i think we have to receive data from the socket inside of the loop as well, and append the data to the data that we already have (hopefully without making any copies here, of course...), don't appear to be doing that at the moment. at least, it would be nice to have a little handler that does an h11 request/response that we could test with some dummy headers and bodies to make sure we receive a whole request for large bodies, can mux the server correctly, keep conns alive, etc